api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -34,117 +42,109 @@ public Cluster apply(org.pmiops.workbench.notebooks.model.Cluster firecloudClust
-    Map<String, String> labels = new HashMap<String, String>();
-    labels.put("all-of-us", "true");
-    labels.put("created-by", userProvider.get().getEmail());
-    firecloudClusterRequest.setLabels(labels);
-    // TODO: Host our extension somewhere.
-    // firecloudClusterRequest.setJupyterExtensionUri("");
-    firecloudClusterRequest.setJupyterUserScriptUri(
-        workbenchConfigProvider.get().firecloud.jupyterUserScriptUri);
-
-    return firecloudClusterRequest;
-  }
-
-  private String convertClusterName(String workspaceId) {
-      String clusterName = workspaceId + this.userProvider.get().getUserId();
-      clusterName = clusterName.toLowerCase();
-      return clusterName;
-  }
+  private final NotebooksService notebooksService;
+  private final Provider<User> userProvider;
+  private final FireCloudService fireCloudService;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   @Autowired
   ClusterController(NotebooksService notebooksService,
       Provider<User> userProvider,
-      WorkspaceService workspaceService,
+      FireCloudService fireCloudService,
       Provider<WorkbenchConfig> workbenchConfigProvider) {
     this.notebooksService = notebooksService;
     this.userProvider = userProvider;
-    this.workspaceService = workspaceService;
+    this.fireCloudService = fireCloudService;
     this.workbenchConfigProvider = workbenchConfigProvider;
   }
 
-  @Override
-  public ResponseEntity<Cluster> createCluster(String workspaceNamespace,
-      String workspaceId) {
-
-    // This also enforces registered auth domain.
-    workspaceService.enforceWorkspaceAccessLevel(workspaceNamespace,
-        workspaceId, WorkspaceAccessLevel.WRITER);
-
-    String clusterName = this.convertClusterName(workspaceId);
-    Cluster createdCluster = TO_ALL_OF_US_CLUSTER.apply(
-        this.notebooksService.createCluster(
-            workspaceNamespace, clusterName, createFirecloudClusterRequest()));
-    return ResponseEntity.ok(createdCluster);
-  }
-
-
-  @Override
-  public ResponseEntity<EmptyResponse> deleteCluster(String workspaceNamespace,
-      String workspaceId) {
-
-    // This also enforces registered auth domain.
-    workspaceService.enforceWorkspaceAccessLevel(workspaceNamespace,
-        workspaceId, WorkspaceAccessLevel.WRITER);
-
-    String clusterName = this.convertClusterName(workspaceId);
-    this.notebooksService.deleteCluster(workspaceNamespace, clusterName);
-    return ResponseEntity.ok(new EmptyResponse());
+  /**
+   * Deterministically produces a cluster name for a given user/environment.
+   * Clusters are namespaced under a free tier project, which is unique to each
+   * user, therefore the only true requirements here are to:
+   * 1. have a reproducible mapping of user -> cluster name (as we currently
+   *    don't store this)
+   * 2. generate a unique name across AoU environments (decision still pending)
```

Comment:
I'm not quite following the second possible requirement here.

If clusters are namespaced under a free tier project, then you should be able to just call every cluster "aou" as far as it goes and it should work... free tier projects aren't shared across environments. (And for that matter, apart from test vs. local, users aren't shared across environments -- we have different GSuite subdomains for each env.)

If clusters weren't namespaced that way, then what you have below won't be guaranteed to be unique and we'd need to come up with something else.